src/main/java/it/gov/pagopa/pu/pagopapayments/mapper/AcaDebtPositionMapper.java

```diff
@@ -0,0 +1,77 @@
+package it.gov.pagopa.pu.pagopapayments.mapper;
+
+import it.gov.pagopa.nodo.paCreatePosition.dto.generated.NewDebtPositionRequest;
+import it.gov.pagopa.pu.p4pa_debt_positions.dto.generated.DebtPositionTypeOrg;
+import it.gov.pagopa.pu.pagopapayments.connector.DebtPositionClient;
+import it.gov.pagopa.pu.pagopapayments.dto.generated.DebtPositionDTO;
+import it.gov.pagopa.pu.pagopapayments.dto.generated.PersonDTO;
+import it.gov.pagopa.pu.pagopapayments.dto.generated.TransferDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.BooleanUtils;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Component
+@Slf4j
+public class AcaDebtPositionMapper {
+
+  private final DebtPositionClient debtPositionClient;
+
+  public AcaDebtPositionMapper(DebtPositionClient debtPositionClient) {
+    this.debtPositionClient = debtPositionClient;
+  }
+
+  public final static OffsetDateTime MAX_DATE = LocalDateTime.of(2099,12,31,23,59,59).atZone(ZoneId.of("Europe/Rome")).toOffsetDateTime();
+
+  public List<NewDebtPositionRequest> mapToNewDebtPositionRequest(DebtPositionDTO debtPosition, Set<String> filterInstallmentStatus, String accessToken) {
+    List<NewDebtPositionRequest> response = new ArrayList<>();
+    final AtomicReference<DebtPositionTypeOrg> debtPositionTypeOrg = new AtomicReference<>();
+
+    debtPosition.getPaymentOptions().forEach(paymentOption ->
+      paymentOption.getInstallments().forEach(installment -> {
```

Comment:
instead of having for loops intended to fill a List, let's use stream API to directly build the list
```suggestion
    return debtPosition.getPaymentOptions().stream()
      .flatMap(paymentOption -> paymentOption.getInstallments().stream())
      .filter(this::installment2skip) // inside this method you should move the if/return logic below
      .map(installment -> {
         ...
         return new request...
         })
         .toList();
```

Reply:
done